tracing-opentelemetry-http/src/main/java/io/micronaut/tracing/opentelemetry/instrument/http/server/MicronautHttpServerNameExtractor.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2017-2022 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.tracing.opentelemetry.instrument.http.server;
+
+import io.micronaut.core.annotation.Internal;
+import io.micronaut.http.HttpRequest;
+import io.opentelemetry.instrumentation.api.instrumenter.SpanNameExtractor;
+
+@Internal
+public enum MicronautHttpServerNameExtractor implements SpanNameExtractor<HttpRequest<Object>> {
```

Comment:
This class is surplus

Reply:
The reason this class exists is that it adds the method to the name (while the `HttpSpanNameExtractor` only uses the path, which does not include the method). Please verify that I understood the [reference](https://opentelemetry.io/docs/reference/specification/trace/api/#span) correctly.

We can create a PR to opentelemetry if you think that it is a better solution and justified.

Reply:
https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/http/HttpSpanNameExtractor.java#L39

I think it uses both method and route. :)

Reply:
Oh, I see. I think I somehow was using an older version - they changed it in v1.23.0. Sorry for the confusion